### PR TITLE
Pac5 calculate size

### PR DESCRIPTION
**Pull request by tomoicki**
Done some refactoring, renaming.
It does work - takes values u put in "Sprawdź rozmiar", sends it to backend, determines if its [Small,Medium,Large,None(error case)] and gives back the answer to frontend.
Ignore classes which start with "Recalculate", they're internally commented - dont need them now but might be useful in the future.
If there's anything to change, please let me know.
Cheers

### CODE REVIEW

File: frontend/src/app/services/post-dimensions.service.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class PostDimensionsService {
+  private PARCELS_URL = 'http://localhost:8080/parcelservice/v1/dimensions';
+  constructor(private httpClient: HttpClient, private logger: NGXLogger) {
+  }
+  postDimensions(dummy: DummyParcelDimensions): Observable<ParcelSizeHolder> {
+    const parcelURL = this.PARCELS_URL;
+    this.logger.info('Chyba zatrybiło?');
```
Comment (TheFlyingKeyboard) [inline frontend/src/app/services/post-dimensions.service.ts]: Please change this log to something more meaningful.

File: frontend/src/app/components/calculate-size/calculate-size.component.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export class CalculateSizeComponent implements OnInit {
+  size: string;
+  constructor(private router: Router, private postDimensionsService: PostDimensionsService, private logger: NGXLogger) {
+  }
+
+  ngOnInit(): void {
+  }
+
+  calcsml(length: number, width: number, height: number): void {
+    const dummyParcelDimensions = new DummyParcelDimensions();
```
Comment (TheFlyingKeyboard) [inline frontend/src/app/components/calculate-size/calculate-size.component.ts]: Please add constructor in DummyParcelDimensions() to accept and set these 3 fields

File: frontend/src/app/common/dummy-parcel-dimensions.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export class DummyParcelDimensions {
```
Comment (TheFlyingKeyboard) [inline frontend/src/app/common/dummy-parcel-dimensions.ts]: Please rename to ParcelDimensions

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSizeDto.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.webappsbusters.parcelmanagement.domain;
+
+public enum ParcelSizeDto {
+    None,
+    Small,
+    Medium,
+    Large
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSizeDto.java]: Enums should have elements in UPPERCASE - NONE, SMALL, ...

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSize.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.webappsbusters.parcelmanagement.domain;
+
+public enum ParcelSize {
+    None,
+    Small,
+    Medium,
+    Large
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSize.java]: Enums should have elements in UPPERCASE - NONE, SMALL, ...

File: backend/src/main/java/com/webappsbusters/parcelmanagement/service/DetermineSize.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.webappsbusters.parcelmanagement.service;
+import com.webappsbusters.parcelmanagement.domain.*;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class DetermineSize {
+    public ParcelSize DetermineDummyParcelSize(int length, int width, int height) {
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/service/DetermineSize.java]: please rename to calculateParcelSize

File: backend/src/main/java/com/webappsbusters/parcelmanagement/controller/DetermineSizeController.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    private final MapperFacade mapperFacade;
+
+    @Autowired
+    public DetermineSizeController(DetermineSize determineSize, MapperFacade mapperFacade) {
+        this.determineSize = determineSize;
+        this.mapperFacade = mapperFacade;
+    }
+
+    @PostMapping("/dimensions")
+    public ResponseEntity<ParcelSizeHolderDto> getDummyParcelDimensions(@RequestBody ParcelDimensionsDto parcelDimensionsDto) {
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/controller/DetermineSizeController.java]: Please rename to calculateParcelDimensions

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensions.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+@Data
+@ToString
+public class ParcelDimensions {
+    int dummywidth;
+    int dummylength;
+    int dummyheight;
+    public int getWidth(Parcel parcel) { return dummywidth = parcel.getWidth(); }
+    public int getLength(Parcel parcel) { return dummylength = parcel.getLength(); }
+    public int getHeight(Parcel parcel) { return dummyheight = parcel.getHeight(); }
+}
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensions.java]: I think those methods are not used, you are using getWidth(), ... generated by Lombok

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensionsDto.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+@NoArgsConstructor
+@Data
+@ToString
+public class ParcelDimensionsDto {
+    int dummywidth;
+    int dummylength;
+    int dummyheight;
+    public int getWidth(Parcel parcel) { return dummywidth = parcel.getWidth(); }
+    public int getLength(Parcel parcel) { return dummylength = parcel.getLength(); }
+    public int getHeight(Parcel parcel) { return dummyheight = parcel.getHeight(); }
```
Comment (TheFlyingKeyboard) [inline backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensionsDto.java]: I think those methods are not used, you are using getWidth(), ... generated by Lombok

File: frontend/src/app/services/post-dimensions.service.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+import {Observable} from 'rxjs';
+import {HttpClient} from '@angular/common/http';
+import {NGXLogger} from 'ngx-logger';
+import {DummyParcelDimensions} from '../common/dummy-parcel-dimensions';
+import {ParcelSizeHolder} from '../common/parcel-size-holder';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PostDimensionsService {
```
Comment (TheFlyingKeyboard) [inline frontend/src/app/services/post-dimensions.service.ts]: Please rename to CalculateParcelDimensionsService

**Review by TheFlyingKeyboard — CHANGES_REQUESTED** [review]

File: frontend/src/app/services/post-dimensions.service.ts (hunk 1 above)
Comment (tomoicki) [reply frontend/src/app/services/post-dimensions.service.ts]: changed to:
this.logger.info('Sending values to {}:', parcelURL);

File: frontend/src/app/components/calculate-size/calculate-size.component.ts (hunk 2 above)
Comment (tomoicki) [reply frontend/src/app/components/calculate-size/calculate-size.component.ts]: done

File: frontend/src/app/common/dummy-parcel-dimensions.ts (hunk 3 above)
Comment (tomoicki) [reply frontend/src/app/common/dummy-parcel-dimensions.ts]: done

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSizeDto.java (hunk 4 above)
Comment (tomoicki) [reply backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSizeDto.java]: done

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSize.java (hunk 5 above)
Comment (tomoicki) [reply backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelSize.java]: done

File: backend/src/main/java/com/webappsbusters/parcelmanagement/service/DetermineSize.java (hunk 6 above)
Comment (tomoicki) [reply backend/src/main/java/com/webappsbusters/parcelmanagement/service/DetermineSize.java]: not sure, whole service name or just the method?

Comment (tomoicki) [reply backend/src/main/java/com/webappsbusters/parcelmanagement/service/DetermineSize.java]: changed the method only

File: backend/src/main/java/com/webappsbusters/parcelmanagement/controller/DetermineSizeController.java (hunk 7 above)
Comment (tomoicki) [reply backend/src/main/java/com/webappsbusters/parcelmanagement/controller/DetermineSizeController.java]: done

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensions.java (hunk 8 above)
Comment (tomoicki) [reply backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensions.java]: yep ,they're not, they are used in the commented classes...i'll comment them as well

File: backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensionsDto.java (hunk 9 above)
Comment (tomoicki) [reply backend/src/main/java/com/webappsbusters/parcelmanagement/domain/ParcelDimensionsDto.java]: same story as with ParcelDimensions

File: frontend/src/app/services/post-dimensions.service.ts (hunk 10 above)
Comment (tomoicki) [reply frontend/src/app/services/post-dimensions.service.ts]: done

**Review by TheFlyingKeyboard — APPROVED** [review]

**Pull request merged**